src/common/snippets/src/lowered/pass/insert_tail_loop.cpp

```diff
@@ -105,21 +281,30 @@ void InsertTailLoop::tail_transformations(LinearIR& linear_ir,
         // correct math calculations for ReduceMax and ReduceSum in scalar case.
         // Note: We find Maximum and Add ops because HorizonMax and HorizonSum are outside Loop,
         //       so they are missed in <tail>
-        auto op = (*expr_it)->get_node();
+        const auto& expr = *expr_it;
+        const auto op = expr->get_node();
         if (config.m_need_fill_tail_register &&
             (ov::is_type<ov::op::v1::Maximum>(op) ||
              ov::is_type<ov::op::v1::Add>(op))) {
             for (size_t i = 0; i < op->inputs().size(); ++i) {
                 if (auto fill = insertFill(op->input(i))) {
-                    const auto& input = expr_it->get()->get_input_port_connector(i);
+                    const auto& input = expr->get_input_port_connector(i);
                     const auto consumers = input->get_consumers();
+                    // If there are several consumers, fill expression must be inserted before first of them
```

Comment:
Could you please elaborate what is the corner case here?
I really don't understand what it should cover.

Reply:
Oh, I see. Thanks a lot for the detailed explanation!